contrib/terraform/vsphere/variables.tf

```diff
@@ -0,0 +1,96 @@
+## Global ##
+
+variable prefix {
+  default = ""
+}
+
+variable "machines" {
+  description = "Cluster machines"
+  type = map(object({
+    node_type = string
+    ip      = string
+  }))
+}
+
+variable inventory_file {
+  default = "inventory.ini"
+}
+
+variable "network" {
+  default = "VM Network"
+}
+
+variable "gateway" {
+  default = ""
+}
+
+variable "dns_primary" {
+  default = ""
+}
+
+variable "dns_secondary" {
+  default = ""
+}
+
+variable "vsphere_datacenter" {
+  default = ""
+}
+
+variable "vsphere_compute_cluster" {
+  default = ""
+}
+
+variable "vsphere_datastore" {
+  default= ""
+}
+
+variable "vsphere_server" {
+  default = ""
+}
+
+variable "vsphere_hostname" {
+  default = ""
+}
+
+variable "firmware" {
+  default = "bios"
+}
+
+variable "hardware_version" {
+  default = "15"
+}
+
+variable "template_name" {
+  default = "ubuntu-focal-20.04-cloudimg"
+}
+
+variable "ssh_pub_key" {
```

Comment:
Please change this to `ssh_public_keys`, which is an inline list of SSH public keys, like with the Exoscale Terraform scripts. This makes it easier for operators to share clusters, since they don't need to worry about where the SSH key is located.